src/gui/sdlmain.cpp

```diff
@@ -655,42 +651,10 @@ void GFX_RequestExit(const bool pressed)
 	}
 }
 
-Bitu GFX_GetBestMode(Bitu flags)
+uint8_t GFX_GetBestMode(uint8_t flags)
 {
-	switch (sdl.desktop.want_type) {
-	case SCREEN_SURFACE:
-	check_surface:
-		switch (sdl.desktop.pixel_format) {
-		case PixelFormat::Indexed8:
-			if (flags & GFX_CAN_8) flags&=~(GFX_CAN_15|GFX_CAN_16|GFX_CAN_32);
-			break;
-		case PixelFormat::BGR555:
-			if (flags & GFX_CAN_15) flags&=~(GFX_CAN_8|GFX_CAN_16|GFX_CAN_32);
-			break;
-		case PixelFormat::BGR565:
-			if (flags & GFX_CAN_16) flags&=~(GFX_CAN_8|GFX_CAN_15|GFX_CAN_32);
-			break;
-		case PixelFormat::BGR888:
-		case PixelFormat::BGRX8888:
-			if (flags & GFX_CAN_32) flags&=~(GFX_CAN_8|GFX_CAN_15|GFX_CAN_16);
-			break;
-		}
-		flags |= GFX_CAN_RANDOM;
-		break;
-#if C_OPENGL
-	case SCREEN_OPENGL:
-#endif
-	case SCREEN_TEXTURE:
-		// We only accept 32bit output from the scalers here
-		if (!(flags & GFX_CAN_32)) {
-			goto check_surface;
-		}
-		flags&=~(GFX_CAN_8|GFX_CAN_15|GFX_CAN_16);
-		break;
-	default:
-		goto check_surface;
-		break;
-	}
+	flags &= GFX_CAN_32;
+	flags &= ~(GFX_CAN_8 | GFX_CAN_15 | GFX_CAN_16);
```

Comment:
Huge cleanup!
Given this functions always does the same thing now (turns on "can-do-32-bit colour" and shuts off the other bits), this need for it is getting quite tenuous.. so next refactoring it could probably just be thrown out (and we simply assume `GFX_CAN_32`).. making the need for all those GFX_CAN flags also unnecessary too.

Reply:
In theory, we could check for 16-bit texture format availability for OpenGL, or other formats for SDL renderers (its software backend supports even indexed formats). But probably I'll be doing that in a different manner, in a different place, if at all.

Reply:
Isn't that micro-optimisation? The most demanding late (S)VGA games are almost all rendered as 32-bit because of the 18-bit RGB palette, no? And it's rather pointless to optimise anything for the very undemanding CGA/EGA/etc. games.

I would actually do the opposite, which is the current behaviour; just treat everything as 32-bit, job done.

I'm a ruthless simplifier, I recommend that to everyone 😄  I only optimise if the gains are significant (more than 10%, in general).

Reply:
Currently that conversion to 32-bits is performed by the CPU. Considering we're constantly short on cycles, feeding raw image data to the GPU whenever possible is far from being a "micro"-optimization (unless you're a rich boi with 13900K + RTX 4090 setup). I'm not talking about the final rendered image though, it's almost always 8- or 10-bits per channel (or float, in case of HDR displays).

Reply:
i7 4790k + RTX 3060 here, bought in 2015 😎 That's was middle of the road hardware even then, 8 years ago, and every DOS game is blazing fast already on my machine. So was on my previous card, a GTX 750 Ti with all emulators, WinUAE, PCSX2, VICE, etc.

So I'm not really buying that it matters, but benchmark it, and then we're talking 😄 

But that's a pretty sane way of working: do the simple thing first, benchmark, then complicate it if it really matters (after benchmarking, not before based on assumptions).

Reply:
Plus, if you're thinking about just passing the palette indexes for 256 colour graphics plus the palette to the GPU (you mentioned this before) as byte values, beware that some games and demos change the palette mid-screen. That's the reason why currently we're rendering all 256-colour modes as 32-bit because that's fiddly to detect whether the palette was changed mid-screen or not.

Most programs don't do this, of course, but you still need to handle this special case... somehow if you want to try this.